### PR TITLE
Add HAProxy image mod & a few minor fixes for router-perf

### PR DESCRIPTION
**Pull request by sjug**
### Description
- Add new HAProxy image deployment
- Cleanup

### CODE REVIEW

File: workloads/router-perf-v2/common.sh [hunk 1]
```diff
@@ -83,18 +95,21 @@ run_mb(){
...
-  oc scale --replicas=1 -n openshift-cluster-version deploy/cluster-version-operator
+  oc patch clusterversions/version --type=json --patch='[{"op":"add","path":"/spec/overrides","value":[{"kind":"Deployment","group":"apps/v1","name":"ingress-operator","namespace":"openshift-ingress-operator","unmanaged":false}]}]'
   oc scale --replicas=1 -n openshift-ingress-operator deploy/ingress-operator
 }
 
 cleanup_infra(){
   log "Deleting infrastructure"
   oc delete ns -l kube-burner-uuid=${UUID} --ignore-not-found
   rm -f /tmp/temp-route*.txt
+  rm http-perf.yml http-*.json
```
Comment (dry923) [inline workloads/router-perf-v2/common.sh:112]: Minor nit but can we combine this w/ the rm line above it?

File: workloads/router-perf-v2/common.sh [hunk 2]
```diff
@@ -36,15 +38,24 @@ deploy_infra(){
...
 
 tune_liveness_probe(){
   log "Disabling cluster version and ingress operators"
-  oc scale --replicas=0 -n openshift-cluster-version deploy/cluster-version-operator
+  oc patch clusterversions/version --type=json --patch='[{"op":"add","path":"/spec/overrides","value":[{"kind":"Deployment","group":"apps/v1","name":"ingress-operator","namespace":"openshift-ingress-operator","unmanaged":true}]}]'
+  if [[ ! -z "${HAPROXY_IMAGE}" ]]; then
+    log "Replacing router with ${HAPROXY_IMAGE}"
+    oc -n openshift-ingress-operator patch deploy/ingress-operator --type=strategic --patch='{"spec":{"template":{"spec":{"containers":[{"name":"ingress-operator","env":[{"name":"IMAGE","value":"'${HAPROXY_IMAGE}'"}]}]}}}}'
+    log "Waiting 60s for ingress-operator IMAGE to change"
+    sleep 60
```
Comment (dry923) [inline workloads/router-perf-v2/common.sh:57]: Is there something we can "wait" for instead of a blind sleep?

Comment (sjug) [reply workloads/router-perf-v2/common.sh]: We can try lol

File: workloads/router-perf-v2/common.sh [hunk 3]
```diff
@@ -36,15 +38,24 @@ deploy_infra(){
...
 
 tune_liveness_probe(){
   log "Disabling cluster version and ingress operators"
-  oc scale --replicas=0 -n openshift-cluster-version deploy/cluster-version-operator
+  oc patch clusterversions/version --type=json --patch='[{"op":"add","path":"/spec/overrides","value":[{"kind":"Deployment","group":"apps/v1","name":"ingress-operator","namespace":"openshift-ingress-operator","unmanaged":true}]}]'
+  if [[ ! -z "${HAPROXY_IMAGE}" ]]; then
+    log "Replacing router with ${HAPROXY_IMAGE}"
+    oc -n openshift-ingress-operator patch deploy/ingress-operator --type=strategic --patch='{"spec":{"template":{"spec":{"containers":[{"name":"ingress-operator","env":[{"name":"IMAGE","value":"'${HAPROXY_IMAGE}'"}]}]}}}}'
+    log "Waiting for ingress-operator IMAGE to change"
+    oc rollout status -n openshift-ingress deploy/router-default
```
Comment (dry923) [inline workloads/router-perf-v2/common.sh:57]: Don't you need a -w or --watch=true on there so it watches to completion or is that the default behavior now?

Comment (sjug) [reply workloads/router-perf-v2/common.sh:57]: I've never needed it, pretty sure `--watch=true` is the default.

**Comment by sjug** [comment]
Wow I didn't know this PR was still open...

**Pull request merged**